skimage/segmentation/_join.py

```diff
@@ -114,20 +116,29 @@ def relabel_sequential(label_field, offset=1):
     >>> relab
     array([5, 5, 6, 6, 7, 9, 8])
     """
+    offset = int(offset)
+    if offset <= 0:
+        raise ValueError("Offset must be strictly positive.")
+    if int(label_field.min()) < 0:
+        raise ValueError("Cannot relabel array that contains negative values.")
     m = label_field.max()
-    if not np.issubdtype(label_field.dtype, np.signedinteger):
+    if not np.issubdtype(label_field.dtype, np.integer):
         new_type = np.min_scalar_type(int(m))
         label_field = label_field.astype(new_type)
         m = m.astype(new_type)  # Ensures m is an integer
     labels = np.unique(label_field)
     labels0 = labels[labels != 0]
-    if m == len(labels0):  # nothing to do, already 1...n labels
+    required_type = np.min_scalar_type(offset + len(labels0))
```

Comment:
min_scalar_type is a little annoying. It can return unsigned types. I think I've personally come around to not liking the use of "unsigned" unless you need specific "unsigned" behavior, that is, `255+2 == 1` is something you want to be true in your math.

Thoughts?

Reply:
The whole `required_type` thing is really covering an edge case that I don't think will happen much in practice, unless someone purposefully uses restricted data types (especially `uint8`).

I frequently save label masks to disk as `uint16` TIFF files. Hence, they have this type when loaded again from disk. I like working with 16-bit integers for space reasons, which can be important for large 3D arrays.

Reply:
Want to allow a dtype parameter to ensure uint16???

Reply:
I think the new logic (use the larger of min_dtype and input dtype) is sufficient here.

Reply:
Sure.